### PR TITLE
2590 Add more pad args in ResizeWithPadOrCrop transform

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2590 .

### Description
This PR added more pad args in the `ResizeWithPadOrCrop` transform.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
This PR is based on user's feedback, it should be a follow-up PR of https://github.com/Project-MONAI/MONAI/pull/2465.

Thanks.

File: monai/transforms/croppad/array.py [hunk 1]
```diff
@@ -967,15 +967,21 @@ class ResizeWithPadOrCrop(Transform):
             ``"median"``, ``"minimum"``, ``"reflect"``, ``"symmetric"``, ``"wrap"``, ``"empty"``}
             One of the listed string values or a user supplied function for padding. Defaults to ``"constant"``.
             See also: https://numpy.org/doc/1.18/reference/generated/numpy.pad.html
+        method: {``"symmetric"``, ``"end"``}
+            Pad image symmetric on every side or only pad at the end sides. Defaults to ``"symmetric"``.
```
Comment (yiheng-wang-nv) [inline monai/transforms/croppad/array.py:971]: Pad image symmetrically
Please also modify the same place in `SpatialPad`

**Review by yiheng-wang-nv — APPROVED** [review]

File: monai/transforms/croppad/array.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/croppad/array.py:971]: Hi @yiheng-wang-nv ,

Thanks for your review.
I think we already have this arg in `SpatialPad`.

Thanks.

Comment (Nic-Ma) [reply monai/transforms/croppad/array.py]: updated in the latest commit.

Thanks.

**Comment by wyli** [comment]
Could you help review all the usage of spatialpad in the codebase for the same issue? It seems we are still missing a few

**Comment by Nic-Ma** [comment]
Hi @wyli ,

OK, let me try to add for all the places in the code base.

Thanks.

**Pull request merged**